fix: remove match/tags from push command

vigneshshanmugam opened the pull request:
+ fix #780
+ fix #787 
+ PR removes the ability to add `--tags`and `--match` to the push command and aborts the command if the command is run with any of the CLI flags.
+ Adds the ability to configure tags for the project based browser and lightweight monitors. Will be overridden by the individual monitor.tags or journey.tags if present on the monitors themselves. 
+ `private_locations` and `privateLocations` are supported in the lightweight monitor config. 
+ `alert.status, alert.tls` was not captured from global config file for browser monitors. Fixed that as well. 


### Testing
- Build the project `npm run build`
- Run the Push command 
```sh
SYNTHETICS_API_KEY=<key> node dist/cli.js push --tags "foo*"
// aborts and logs the error
```
- Test with configuring `monitor.tags` in the synthetics.config file and check if its applied on all monitors in the UI.

devcorpio commented on src/options.ts:
```diff
@@ -120,22 +120,33 @@ export function normalizeOptions(
-        options.privateLocations ?? monitor?.privateLocations;
-      options.alert = options.alert ?? monitor?.alert;
+      for (const key of Object.keys(monitor || {})) {
+        // TODO: screenshots require special handling as the flags are different
```
Does this TODO mean that you will revisit this area later on?

vigneshshanmugam replied:
Nope, I will remove TODO here.

devcorpio reviewed (commented):
**I believe something is wrong with the push command and the tags**

My understanding is that tags will never filter out monitors when pushing, regardless of where those tags are defined (synthetics.config.file, monitor.use.tag, and journey({...})

**Scenario tested:**

I defined a tag within the **synthetics.config.ts** file

<img width="472" alt="Screenshot 2023-07-06 at 12 02 42" src="https://github.com/elastic/synthetics/assets/15065076/a9024384-2b6c-4a29-82ca-06ec78935a37">

These are the monitors I wanted to push (**4** monitors):

<img width="288" alt="Screenshot 2023-07-06 at 12 01 08" src="https://github.com/elastic/synthetics/assets/15065076/566c6050-9d72-40b9-a6f1-99a5e5f65b3e">

After that, I executed the push command:

```SYNTHETICS_API_KEY=mykey npm run push```

My surprise here has been to see that **just one monitor** has been pushed. The only one that has been pushed is the lightweight one (the tag was also visible in the UI). My expectations were to see the 4 monitors pushed and visible in the UI with the corresponding tag.

The filtering (that just affect browser monitors) takes place [here](https://github.com/elastic/synthetics/blob/main/src/core/runner.ts#L403)

This is the output of my logs (you can see 3 entries there)

<img width="462" alt="Screenshot 2023-07-06 at 12 12 17" src="https://github.com/elastic/synthetics/assets/15065076/477831a5-6631-4d26-8ef9-76cbc5c52954">


---

On the other hand, things that I can confirm that work greatly:




--

`SYNTHETICS_API_KEY=mylocalkey npm run push -- --match 'foo*'`

**Output**:

<img width="401" alt="Screenshot 2023-07-06 at 11 24 41" src="https://github.com/elastic/synthetics/assets/15065076/4ac4e708-06d0-425c-8e39-a16c8d26b293">


--

`SYNTHETICS_API_KEY=mylocalkey npm run push -- --tags 'baz'`

**Output**:

<img width="401" alt="Screenshot 2023-07-06 at 11 24 41" src="https://github.com/elastic/synthetics/assets/15065076/4ac4e708-06d0-425c-8e39-a16c8d26b293">

--

> Test with configuring monitor. tags in the synthetics.config file and check if it's applied on all monitors in the UI.

<img width="1122" alt="Screenshot 2023-07-06 at 11 27 54" src="https://github.com/elastic/synthetics/assets/15065076/b4b355b1-cebf-4ba4-aa0b-55ca31f29ddd">


--
> Will be overridden by the individual monitor.tags or journey.tags if present on the monitors themselves.

**Scenario**: tag defined in **synthetics.config.file** + **journey** 

The tag defined in the **journey** wins.

**Scenario**: tag defined in **synthetics.config.file** + **journey** + **monitor.use.tag**

The tag defined in the **monitor.use.tag** wins.

vigneshshanmugam commented:
> My surprise here has been to see that just one monitor has been pushed. The only one that has been pushed is the lightweight one (the tag was also visible in the UI). My expectations were to see the 4 monitors pushed and visible in the UI with the corresponding tag.

Good catch @devcorpio , Fixed it now and added test to capture this. Also found another bug with alerts not being added to global config.

devcorpio reviewed (approved):
LGTM! 🎉  everything works now
